File: NumberOf1Between1AndN_Solution.py
# -*- coding:utf-8 -*-
class Solution:
    def NumberOf1Between1AndN_Solution(self, n):
        # write code here
        result = 0
        for value in range(1,n+1):
            ret = str(value)
            result += ret.count('1')
            
        return result

File: NumberOf1Between1AndN_Solution.java
public class Solution {
    public int NumberOf1Between1AndN_Solution(int n) {
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i <= n; i++){
            sb.append(i);
        }
        int count = 0;
        int length = sb.length();
        for (int i = 0; i < length; i++){
            if (sb.charAt(i) == '1')
                count++;
        }
        return count;
    }
}